### PR TITLE
Upgrade deprecated resources

### PR DESCRIPTION
**Pull request by jasiu001**
**Description**

Changes proposed in this pull request:

- removal deprecated API
- moving ServiceCatalog custom resource definitions to the cluster-essentials component

**Related issue(s)**
#8563 
#7933

### CODE REVIEW

**Comment by m00g3n** [comment]
LGTM

File: resources/cluster-essentials/templates/crd-init-service-catalog.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  servicebindings.servicecatalog.k8s.io: |-
+{{.Files.Get "files/crd-servicebinding.servicecatalog.yaml" | printf "%s" | indent 4}}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-init-service-catalog
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    helm.sh/hook-delete-policy: before-hook-creation, hook-succeeded"
```
Comment (mszostok) [inline resources/cluster-essentials/templates/crd-init-service-catalog.yaml:34]:
```suggestion
    helm.sh/hook-delete-policy: "before-hook-creation, hook-succeeded"
```

File: resources/cluster-essentials/templates/crd-init-service-catalog.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                clusterserviceclasses.servicecatalog.k8s.io \
+                clusterserviceplans.servicecatalog.k8s.io \
+                servicebrokers.servicecatalog.k8s.io \
+                serviceclasses.servicecatalog.k8s.io \
+                serviceplans.servicecatalog.k8s.io \
+                serviceinstances.servicecatalog.k8s.io \
+                servicebindings.servicecatalog.k8s.io )
+
+              for crd in "${crds[@]}"; do
+                timeout 60s bash -c "until kubectl get $crd || kubectl apply -f /etc/crd/$crd; do sleep 2; done"
```
Comment (mszostok) [inline resources/cluster-essentials/templates/crd-init-service-catalog.yaml:75]: bash... so we will never learn to do it right
- no testing 
- no error handling (always apply with 2s sleep unit 60s timeout 🤦 ) IMO it's good for POC but not for production
- strange timeout and sleep

Comment (jasiu001) [reply resources/cluster-essentials/templates/crd-init-service-catalog.yaml:75]: I suggest create issue about all "init jobs" and mark this PR as blocked. What do you mean?

Comment (jasiu001) [reply resources/cluster-essentials/templates/crd-init-service-catalog.yaml:75]: After discussion with the author of jobs functionality, common logic for all init-jobs will be extracted and rewritten in the next, second stage.

**Pull request merged**